### DISCUSSION TITLE
Remove width / height attribute from svg

### DISCUSSION THREAD

**Discussion by Emporea** [observablehq/plot#728] [Q&A] [answered]
I append my plots with element.appentChild.
`document.getElementById("div").appendChild(Plot.plot());`

This way the svg automatically gets the attributes height and width.

I dont talk about `Plot.plot({width: x, height: y...})`

I mean this
![image](https://user-images.githubusercontent.com/21969523/151675071-26d606b2-cb39-4e0e-a65e-e352378a2288.png)

This heavily interferes with scaling capabilities. In a grid it just doesn't fit and doesnt resize all the time.
I simply fix this this by removing them by hand everytime:
```
svg_element.removeAttribute("height");
svg_element.removeAttribute("width");
```

Is there a way in Observable Plot to disable this auto svg attribute setting?

Edit:
To give an example.

With height / width attributes:

![image](https://user-images.githubusercontent.com/21969523/151675201-47128dd2-6ed7-449e-ac19-348198d159d8.png)

After removing heigth / width attributes (what I want):

![image](https://user-images.githubusercontent.com/21969523/151675205-b3d11f87-2bcc-4a01-835c-2bfe88c0d501.png)

**Comment by mbostock** [comment, marked as answer]
No, there is not. The top-level width and height options set both the width and height attributes and the viewBox. If you don’t want these attributes you’ll need to remove them yourself.
